lib/puppet/file_serving/metadata.rb

```diff
@@ -138,7 +138,10 @@ def initialize(path,data={})
     @source      = data.delete('source')
     @content_uri = data.delete('content_uri')
 
-    super(path, data.map { |k, v| [k.to_sym, v] }.to_h)
+    links = data.fetch('links', nil) || data.fetch(:links, nil)
+    relative_path = data.fetch('relative_path', nil) || data.fetch(:relative_path, nil)
+    source = @source || data.fetch(:source, nil)
```

Comment:
This isn't the cleanest, but it works...

Reply:
ah 👍 It's interesting how making this explicit shows how confusing it is for callers, to pass a string for `owner`, but symbol or string for `links`).

Reply:
yeah... I had it as strings originally, but the tests failed. So, symbols it is. I'm not confident that `owner` is necessarily a string, but it's working as is, so...